boa/src/builtins/json/mod.rs

```diff
@@ -65,10 +67,70 @@ pub fn parse(_: &mut Value, args: &[Value], _: &mut Interpreter) -> ResultValue
 ///
 /// [spec]: https://tc39.es/ecma262/#sec-json.stringify
 /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify
-pub fn stringify(_: &mut Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
-    let obj = args.get(0).expect("cannot get argument for JSON.stringify");
-    let json = obj.to_json().to_string();
-    Ok(Value::from(json))
+pub fn stringify(_: &mut Value, args: &[Value], interpreter: &mut Interpreter) -> ResultValue {
+    let object = args.get(0).expect("cannot get argument for JSON.stringify");
+
+    if args.len() == 1 {
+        return Ok(Value::from(object.to_json().to_string()));
+    }
+
+    let replacer = args.get(1).expect("cannot get JSON.stringify replacer");
+    if !replacer.is_object() {
+        return Ok(Value::from(object.to_json().to_string()));
+    }
+
+    let replacer_as_object = replacer
+        .as_object()
+        .expect("JSON.stringify replacer was an object");
+    if replacer_as_object.is_callable() {
+        object
+            .as_object()
+            .map(|obj| {
+                let object_to_return = Value::new_object(None);
+                for (key, val) in obj.properties.iter().filter(|(_, v)|v.value.is_some()) {
```

Comment:
Since we are only using the `v.value` here, and it must be `Some()`, I would suggest using `.filter_map()` instead of `.filter()` and returning `Some((key, value))` with an `if let Some(value) = v.value`, and `None` otherwise. That way we can just call `value.clone()` when passing the third parameter to `call()` below.